File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        
        #use the array itself to register the frequency of each array
        size = len(nums)
        solution = []
        for i in range(size):
            if(nums[abs(nums[i])-1] < 0):
                solution.append(abs(nums[i]))
            nums[abs(nums[i])-1] *= -1
            
        return solution

File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) 
    {
        ArrayList<Integer> solution = new ArrayList<Integer>();
        int arrayLen = nums.length;
        
        for(int i = 0;i<arrayLen;i++)
        {
            if(nums[Math.abs(nums[i])-1] < 0)
                solution.add(Math.abs(nums[i]));
            nums[Math.abs(nums[i])-1] *= -1;
        }
    
        return solution;
    }
}